warehouse/accounts/forms.py

```diff
@@ -562,11 +574,11 @@ def validate_username_or_email(self, field):
         if "@" in field.data:
             # Additional checks for the validity of the address
             try:
-                Address(addr_spec=field.data)
-            except (IndexError, ValueError, HeaderParseError):
+                email_validator.validate_email(field.data, check_deliverability=True)
```

Comment:
This doesn't seem like the right place to do this. Wouldn't this block someone from resetting their password if their domain or MX domain is blocked?

Reply:
Oh, or is this just additional validation of the email along the way. :)

Reply:
This is changing the validation to be uniform with registration, as well as prevent someone from setting an seemingly valid-appearing email address that has no MX records, meaning they effectively cut off that email address from use, since it can never be verified.